RInAppMessaging/Classes/Models/Responses/DisplayPermissionResponse.swift

```diff
@@ -1,4 +1,33 @@
-internal struct DisplayPermissionResponse: Decodable {
+internal struct DisplayPermissionResponse: Codable, Equatable {
+
+    private enum CodingKeys: String, CodingKey {
+        case display
+        case performPing
+        case creationTimeMilliseconds
+    }
+
     let display: Bool
     let performPing: Bool
+    let creationTimeMilliseconds: Int64 // Cache coding only
+
+    // For unit tests
+    init(display: Bool, performPing: Bool) {
+        self.display = display
+        self.performPing = performPing
+        creationTimeMilliseconds = Date().millisecondsSince1970
+    }
+
+    init(from decoder: Decoder) throws {
```

Comment:
If the below function calls can throw, should we be wrapping these in a catch?

Reply:
no. The responsibility is passed to the caller of this init :) which is marked as `throws`

Reply:
ok 🙂 

and is the caller the `try JSONDecoder().decode(DisplayPermissionResponse.self` that's wrapped in a do-catch?

Reply:
Yes 🙂